SU2_CFD/src/solvers/CTurbSASolver.cpp

```diff
@@ -301,8 +301,65 @@ void CTurbSASolver::Viscous_Residual(const unsigned long iEdge, const CGeometry*
   };
 
   /*--- Now instantiate the generic implementation with the functor above. ---*/
+  const bool implicit = (config->GetKind_TimeIntScheme() == EULER_IMPLICIT);
+    CFlowVariable* flowNodes = solver_container[FLOW_SOL] ?
+        su2staticcast_p<CFlowVariable*>(solver_container[FLOW_SOL]->GetNodes()) : nullptr;
+
+  /*--- Points in edge ---*/
+  auto iPoint = geometry->edges->GetNode(iEdge, 0);
+  auto jPoint = geometry->edges->GetNode(iEdge, 1);
+
+  /*--- Helper function to compute the flux ---*/
+  auto ComputeFlux = [&](unsigned long point_i, unsigned long point_j, const su2double* normal) {
+    numerics->SetCoord(geometry->nodes->GetCoord(point_i),geometry->nodes->GetCoord(point_j));
+    numerics->SetNormal(normal);
+
+    if (flowNodes) {
+      numerics->SetPrimitive(flowNodes->GetPrimitive(point_i), flowNodes->GetPrimitive(point_j));
+    }
+
+    numerics->SetScalarVar(nodes->GetSolution(point_i), nodes->GetSolution(point_j));
+    numerics->SetScalarVarGradient(nodes->GetGradient(point_i), nodes->GetGradient(point_j));
 
-  Viscous_Residual_impl(SolverSpecificNumerics, iEdge, geometry, solver_container, numerics, config);
+    return numerics->ComputeResidual(config); 
+  };
+
+  SolverSpecificNumerics(iPoint, jPoint);
+
+  /*--- Compute fluxes and jacobians i->j ---*/
+  const su2double* normal = geometry->edges->GetNormal(iEdge);
+  auto residual_ij = ComputeFlux(iPoint, jPoint, normal);
+  // if (ReducerStrategy) {
+  //   EdgeFluxes.SubtractBlock(iEdge, residual_ij);  
+  //   if (implicit) {
+  //     Jacobian.UpdateBlocksSub(iEdge, residual_ij.jacobian_i, residual_ij.jacobian_j);
+  //   }
+  // }
+  // else {
+    LinSysRes.SubtractBlock(iPoint, residual_ij);  
+    if (implicit) {
+      Jacobian.UpdateBlocksSub(iEdge, iPoint, jPoint, residual_ij.jacobian_i, residual_ij.jacobian_j);
+    // }
+  }
+
+  /*--- Compute fluxes and jacobians j->i ---*/
+  su2double flipped_normal[3];
+  for (int iDim=0; iDim<nDim; iDim++)
+    flipped_normal[iDim] = -normal[iDim];
+
+  auto residual_ji = ComputeFlux(jPoint, iPoint, flipped_normal);
+  // if (ReducerStrategy) {
+  //   EdgeFluxes.SubtractBlock(iEdge, residual_ji);  
+  //   if (implicit) {
+  //     Jacobian.UpdateBlocksSub(iEdge, residual_ji.jacobian_i, residual_ji.jacobian_j);
+  //   }
+  // }
+  // else {
+    LinSysRes.SubtractBlock(jPoint, residual_ji);  
+    if (implicit) {
+      Jacobian.UpdateBlocksSub(iEdge, iPoint, jPoint, residual_ji.jacobian_i, residual_ji.jacobian_j);
+    // }
```

Comment:
You are updating the Jacobians, each of the two calls to UpdateBlocksSub is updating the blocks corresponding to ii, ij, ji, jj.
So you are basically doubling the Jacobians.

We don't have an interface to update just a particular row of the matrix. So you can either:
- Divide the Jacobians by 2 in the numerics class so that the two calls result in an average Jacobian
- Compute an average Jacobian in the numerics and use only one call to UpdateBlocksSub
- Do manual updates of ii, ij, and then ji, jj (see `CFVMFlowSolverBase<V, FlowRegime>::BC_Sym_Plane` for an example).